### PR TITLE
Add py35 py36-dev tests, remove failing 2.6 test, move regex compilation for nice speedup

### PR DESCRIPTION
**Pull request by ssteinerx**
As per PR title.  See commit msgs for a little more detail.

### CODE REVIEW

**Comment by ssteinerx** [comment]
Also updated .gitignore to skip .egg-info dir and switched to setuptools from distutils to allow use of:

```
python setup.py develop
```

and all the other attendant improvements from setuptools.

**Comment by ssteinerx** [comment]
This passes tests but does not successfully parse the output from ping on my OS X system.

Will open new one when I've fixed whatever is causing that issue, too.

NOTE: the original (pulled from master of this repo) also fails on my OS X system.

**Comment by gg** [comment]
Thanks for your contributions. Sorry, I haven't been able to test this on OS X. I wrote this as a quick-and-dirty way to parse ping results for myself on Linux and familiarize myself with Python packaging.

I should update the README to direct people to the [pingparsing library](https://github.com/thombashi/pingparsing) as it's more comprehensive and better maintained than my library. If you're looking to contribute, I think it's more worthwhile to test/add OS X compatibility to the pingparsing library.

**Comment by ssteinerx** [comment]
I just whacked the heck out of it.  There is now a place for test data i.e. the output of the ping command with the expected result as the first line and it's in tox.

I'll go look at pingparsing and add my tests there as well.

**Comment by ssteinerx** [comment]
Doesn't look like they tested OS X either.

**Comment by ssteinerx** [comment]
No worries.  

I have need for a ping in a product I'm writing and had to have thorough tests to which I can add data from any systems that cause exceptions anyway so no wasted effort.

**Comment by ssteinerx** [comment]
Just took a look at pingparsing.

It's not more comprehensive, it's just more complicated and unnecessarily so.  I find the code very un-Pythonic and I don't like pyparsing over regular expressions for something this simple.

If I have to extend one, I'll extend this rather than work on that.

File: src/pingparser.py [hunk 1]
```diff
@@ -73,39 +126,42 @@ def main(argv=sys.argv):
...
-    elif args[0].startswith('+'):
-        args[0] = args[0].lstrip('+')
-        output = ' '.join(args[0:])
-    else:
-        parser.print_help()
-
-    for (fmt, rep) in format_replacements:
-        output = output.replace(fmt, rep)
-
+    format_string = default_format
```
Comment (gg) [inline src/pingparser.py]: Custom formatting is not working. I think this should be:

```python
    format_string = options.format or default_format
```

**Review by gg — CHANGES_REQUESTED** [review]
Thanks, I appreciate the improvements! I'm ready to merge this in after you commit a quick fix for custom formatting.

**Comment by gg** [comment]
> It's not more comprehensive, it's just more complicated and unnecessarily so. I find the code very un-Pythonic and I don't like pyparsing over regular expressions for something this simple.

Ah, I didn't bother to dig into the code. After a cursory glace, it appeared to have more features and better documentation/tests. Like you said, no worries!

**Comment by ssteinerx** [comment]
I may have made a mistake there -- there isn't a set of tests for that and I'm interested in using this as a library so my focus wasn't on the command line utility, but on the backend functionality.

You could practice and write a test, or I can get to it maybe tomorrow -- right now I'm in the middle of a live site backend credit card processing nightmare on a site whose busiest days of the year are this week (of course).

S

File: src/pingparser.py (hunk 1 above)
Comment (ssteinerx) [reply src/pingparser.py]: Done.

**Review by gg — APPROVED** [review]

**Comment by gg** [comment]
Thanks again!

**Comment by ssteinerx** [comment]
That'd probably be enough for a version bump and release, yes?

**Comment by gg** [comment]
> That'd probably be enough for a version bump and release, yes?

I bumped the version to 0.4 in setup.cfg.

As far as a release goes... Do you mind if I transfer repository ownership to you? Admittedly, I haven't had use for this project in *years*. I think it's in better off in your hands since you're using it and may make additional contributions in the future.

**Comment by ssteinerx** [comment]
Sure, pass it on (though I'm not sure how to do that).

**Comment by ssteinerx** [comment]
I'm not sure how to xfer the PyPi listing either...

**Comment by gg** [comment]
> Sure, pass it on (though I'm not sure how to do that).

Before I can transfer the repo, you need to rename or remove [your fork of `pingparser`](https://github.com/ssteinerx/pingparser) since it has the same name as this repository.

> I'm not sure how to xfer the PyPi listing either...

AFAIK, pingparser isn't listed on PyPi.

**Comment by ssteinerx** [comment]
My fork is renamed.

**Comment by gg** [comment]
Sorry, it looks like you have to remove the fork. I'm getting this error message:

> ssteinerx already has a repository in the gg/pingparser network

For reference, I found a help page on repository transfers: https://help.github.com/articles/about-repository-transfers/

**Comment by ssteinerx** [comment]
Gone!

**Comment by gg** [comment]
Transfer request sent.

**Pull request merged**